jepsen/src/jepsen/nemesis/combined.clj

```diff
@@ -249,6 +263,133 @@
+                                     :variance     (-> op :value :variance)
+                                     :distribution (-> op :value :distribution)})
+                  :flaky (net/flaky! (:net test) (assoc test :nodes (-> op :value :targets)))
+                  :heal  (net/heal!  (:net test) test)
+                  :fast  (net/fast!  (:net test) test))]
+        (assoc op :value res)))
+
+    (teardown! [_this _test])))
+
+(defn wan-generators
+  "A map with a :generator and :final-generator for WAN-related operations
+  Options are from nemesis-package."
+  [opts]
+  (let [db      (:db opts)
+        faults  (:faults opts)
+        drop?   (contains? faults :drop)
+        slow?   (contains? faults :slow)
+        flaky?  (contains? faults :flaky)
+
+        ;; List of possible specifications for nodes to make slow or flaky. Note
+        ;; that drop is special and already has a source and target node to target.
+        slow-targets  (:targets (:slow  opts (node-specs db)))
+        flaky-targets (:targets (:flaky opts (node-specs db)))
+
+        ;; Drop and heal.
+        drop' (fn [_ _]
+                (let [nodes (-> opts :test :nodes)
+                      src   (rand-nth nodes)
+                      dest  (rand-nth (remove #{src} nodes))]
+                  {:type :info
+                   :f    :drop
+                   :value {:src  src
+                           :dest dest}}))
+        heal {:type :info, :f :heal}
+        drop-heal (gen/flip-flop drop' heal)
+
+        ;; Slow and fast.
+        slow (fn [test _process]
+               (let [mean         50
+                     variance     (rand-nth #{30 50 100 150 1000})
+                     distribution :normal
+                     target-specs (:targets (:slow opts) (node-specs db))
+                     targets      (db-nodes test db
+                                            (some-> target-specs seq rand-nth))]
+                 {:type :info
+                  :f     :slow
+                  :value {:mean         mean
+                          :variance     variance
+                          :distribution distribution
+                          :targets      targets}}))
+        fast {:type :info, :f :fast}
+        slow-fast (gen/flip-flop slow fast)
+
+        ;; Flaky and fast.
+        flaky (fn [test _process]
+                (let [target-specs (:targets (:flaky opts) (node-specs db))
+                      targets (db-nodes test db
+                                        (some-> target-specs seq rand-nth))]
+                  {:type :info
+                   :f    :flaky
+                   :value {:targets targets}}))
+        flaky-fast (gen/flip-flop flaky fast)
+
+        modes (cond-> []
+                drop?  (conj drop-heal)
+                slow?  (conj slow-fast)
+                flaky? (conj flaky-fast))
+
+        final (cond-> []
+                drop?             (conj heal)
+                (or slow? flaky?) (conj fast))]
```

Comment:
These are nice to have, but I think, again, I'd be inclined to split slow/fast, and flaky/perfect, into their own nemesis packages, and make datacenter support a matter of *targets* which could apply to all of them.